recipes/libevent/all/conanfile.py

```diff
@@ -0,0 +1,113 @@
+    exports_sources = ["CMakeLists.txt"]
+    settings = "os", "compiler", "build_type", "arch"
+    options = {"shared": [True, False],
+               "fPIC": [True, False],
+               "with_openssl": [True, False],
+               "disable_threads": [True, False]}
+    default_options = {"shared": False,
+                       "fPIC": True,
+                       "with_openssl": True,
+                       "disable_threads": False}
+    generators = "cmake"
+    _source_subfolder = "source_subfolder"
+
+    def config_options(self):
+        if self.settings.os == "Windows":
+            del self.options.fPIC
+
+    def configure(self):
+        del self.settings.compiler.libcxx
+        del self.settings.compiler.cppstd
+        if self.options.with_openssl and self.options.shared:
+            # static OpenSSL cannot be properly detected because libevent picks up system ssl first
+            # so enforce shared openssl
+            self.output.warn("Enforce shared OpenSSL for shared build")
+            self.options["openssl"].shared = self.options.shared
+
+    def requirements(self):
+        if self.options.with_openssl:
+            self.requires.add("openssl/1.1.1d")
+
+    def source(self):
+        tools.get(**self.conan_data["sources"][self.version])
+        extracted_folder = "libevent-{0}-stable".format(self.version)
+        if self.version == '2.1.11':
+            # temporary fix due to missing files in dist package for 2.1.11, see upstream bug 863
+            extracted_folder = "libevent-release-{0}-stable".format(self.version)
+        os.rename(extracted_folder, self._source_subfolder)
+
+        os.rename(os.path.join(self._source_subfolder, "CMakeLists.txt"),
+                  os.path.join(self._source_subfolder, "CMakeListsOriginal.txt"))
+        shutil.copy("CMakeLists.txt",
+                    os.path.join(self._source_subfolder, "CMakeLists.txt"))
+
+        # patch 'beta' to 'stable' because there is no git repository which cmake uses to determine stage name
+        # needed for 2.1.10, not needed for 2.1.11 so marked as non-strict
+        tools.replace_in_file(os.path.join(self._source_subfolder, "cmake", "VersionViaGit.cmake"),
+                              'set(EVENT_GIT___VERSION_STAGE "beta")',
+                              'set(EVENT_GIT___VERSION_STAGE "stable")',
+                              strict=False)
+
+    def imports(self):
+        # Copy shared libraries for dependencies to fix DYLD_LIBRARY_PATH problems
+        #
+        # Configure script creates conftest that cannot execute without shared openssl binaries.
+        # Ways to solve the problem:
+        # 1. set *LD_LIBRARY_PATH (works with Linux with RunEnvironment
+        #     but does not work on OS X 10.11 with SIP)
+        # 2. copying dylib's to the build directory (fortunately works on OS X)
+
+        if self.settings.os == "Macos":
+            self.copy("*.dylib*", dst=self._source_subfolder, keep_path=False)
```

Comment:
@SSE4 The new feature for OSX Framework is not enough to solve this?

Reply:
need to check, for me appears like nothing about framework, but about DYLD_LIBRARY_PATH

Reply:
agreed, it was a problem of detecting OpenSSL with autotools on OSX. Unfortunately, this problem can be only seen for specific machine configurations. A few homebrew libraries dropped in /usr/local can easily hide the problem. As it thoroughly tested at Bincrafters infrastructure with a ton of build configurations and checking logs, I prefer to keep this workaround.